gaphor/core/modeling/properties.py

```diff
@@ -912,6 +950,8 @@ def __init__(
         self.original: association | derived = original
         self.upper = original.upper
         self.lower = original.lower
+        if opposite is not None:
+            self.original.opposite = opposite
```

Comment:
We should probably not change the original opposite relation when a relation is redefined.

I assume the opposite relation should wither point to the original opposite end, or a redefined one.